### PR TITLE
Fix ESP32 storage

### LINKED ISSUE #8569: ESP32: All clusters app fails to boot up.
#### Problem
* All clusters app for esp32 fails to boot up, it throws below error
```
I (778) wifi:wifi driver task: 3ffdaa8c, prio:23, stack:6656, core=0
I (778) system_api: Base MAC address is not set
I (778) system_api: read default base MAC address from EFUSE
I (788) wifi:wifi firmware version: c7d0450
I (788) wifi:wifi certification version: v7.0
I (788) wifi:config NVS flash: enabled
I (788) wifi:config nano formating: disabled
I (808) wifi:Init data frame dynamic rx buffer num: 32
I (808) wifi:Init management frame dynamic rx buffer num: 32
I (818) wifi:Init management short buffer num: 32
I (818) wifi:Init dynamic tx buffer num: 32
I (818) wifi:Init static rx buffer size: 1600
                                                                                                                                                       I (828) wifi:Init static rx buffer num: 10
I (828) wifi:Init dynamic rx buffer num: 32
I (838) wifi_init: rx ba win: 6
I (838) wifi_init: tcpip mbox: 32
I (838) wifi_init: udp mbox: 6
I (848) wifi_init: tcp mbox: 6
I (848) wifi_init: tcp tx win: 5744
I (848) wifi_init: tcp rx win: 5744
I (848) wifi_init: tcp mss: 1440
I (858) wifi_init: WiFi IRAM OP enabled
I (868) wifi_init: WiFi RX IRAM OP enabled
E (878) chip[DL]: Configuration Manager initialization failed: Error 0x05001102
> E (888) all-clusters-app: device.Init() failed: Error 0x05001102
```

### LINKED ISSUE #8572: ESP32 storage probably broken by commit 162366f0 PR #8469
#### Problem

```
V (11206) chip[IN]: New secure session created for device 0x0000000000000000, key 17!!
I (11216) chip[SVR]: Device completed SPAKE2+ handshake
I (11226) chip[SVR]: Persisting admin ID 0, next available 1
I (11276) CHIP[DL]: Write request received for CHIPoBLE RX characteristic con 0 12
I (11416) CHIP[DL]: Write request received for CHIPoBLE RX characteristic con 0 12
I (11786) chip[SVR]: Stored value in server storage
E (11786) chip[DIS]: Error occurred calling SyncSetKeyValue: Error 0x05001104
```

 #### Proposed Solution

Will fix or revert today.

### PR DESCRIPTION
**Pull request by kpschoedel**
#### Problem

Commit 162366f0 (#8469) broke persistent storage on ESP32.

#### Change overview

- Fixed return value of `ESP32Config::EnsureNamespace()`.
- Fixed a copy-paste mistake (`NVS_READONLY` for `NVS_READWRITE`)
  in `KeyValueStoreManagerImpl::_Put()`.
- Added string presentation of ESP32 errors to aid troubleshooting.

Fixes #8572 _ESP32 storage probably broken_
Fixes #8569 _ESP32: All clusters app fails to boot up._

#### Testing

Manual on M5Stack

### CODE REVIEW

**Review by mspang — APPROVED** [review]

File: src/platform/ESP32/ESP32Utils.cpp [hunk 1]
```diff
@@ -313,3 +313,43 @@ CHIP_ERROR ESP32Utils::MapError(esp_err_t error)
     }
     return ChipError::Encapsulate(ChipError::Range::kPlatform, error);
 }
+
+/**
+ * Given an OpenChip error value that represents an OpenThread error, returns a
```
Comment (bzbarsky-apple) [inline src/platform/ESP32/ESP32Utils.cpp:318]: What is OpenChip?  And what does OpenThread have to do with this?

Should probably fix the OpenChip in `src/platform/OpenThread/OpenThreadUtils.cpp` too.

File: src/platform/ESP32/ESP32Utils.cpp [hunk 2]
```diff
@@ -313,3 +313,43 @@ CHIP_ERROR ESP32Utils::MapError(esp_err_t error)
...
+        return false;
+    }
+
+#if CHIP_CONFIG_SHORT_ERROR_STR
+    const char * desc = NULL;
+#else  // CHIP_CONFIG_SHORT_ERROR_STR
+    const char * desc = esp_err_to_name((esp_err_t) ChipError::GetValue(err));
+#endif // CHIP_CONFIG_SHORT_ERROR_STR
+
+    chip::FormatError(buf, bufSize, "OpenThread", err, desc);
```
Comment (bzbarsky-apple) [inline src/platform/ESP32/ESP32Utils.cpp:342]: Not OpenThread....

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/ESP32/ESP32Utils.cpp (hunk 1 above)
Comment (kpschoedel) [reply src/platform/ESP32/ESP32Utils.cpp:318]: I missed that one; it's copied from something that originally said OpenWeave and went through an s/Weave/Chip/.

File: src/platform/ESP32/ESP32Utils.cpp [hunk 3]
```diff
@@ -315,15 +315,15 @@ CHIP_ERROR ESP32Utils::MapError(esp_err_t error)
 }
 
 /**
- * Given an OpenChip error value that represents an OpenThread error, returns a
+ * Given a CHP error value that represents an ESP32 error, returns a
```
Comment (bzbarsky-apple) [inline src/platform/ESP32/ESP32Utils.cpp:318]:
```suggestion
 * Given a CHIP error value that represents an ESP32 error, returns a
```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by mspang** [comment]
Marking hotfix as this breaks our smoke tests..

@jepenven-silabs @msandstedt @Damian-Nordic

**Pull request merged**